react/src/tests/opReturn.test.ts

```diff
@@ -85,94 +107,109 @@ describe('getDataPushdata', () => {
-    const fullResult = parseOpReturnProps(undefined, true)
-    expect(fullResult).toBe(allResultsPrefix + '00') // 00 is pushdata for no data
-  })
+    const fullResult = encodeOpReturnProps(undefined, true);
+    expect(fullResult).toBe(allResultsPrefix + '00' + '00'); // the 00s are pushdata for the data and paymentId respectively
+  });
   it('Empty data', () => {
-    const fullResult = parseOpReturnProps('')
-    const resultPaymentId = fullResult?.slice(-paymentIdDigits)
-    const resultNoPaymentId = fullResult?.slice(0, -paymentIdDigits)
-    expect(resultNoPaymentId).toBe(allResultsPrefix + '00') // 00 is pushdata for no data
-    expect(resultPaymentId).toMatch(paymentIdRegex)
-  })
+    const fullResult = encodeOpReturnProps('');
+    const resultPaymentId = fullResult?.slice(-paymentIdDigits);
+    const resultNoPaymentId = fullResult?.slice(0, -paymentIdDigits);
+    expect(resultNoPaymentId).toBe(allResultsPrefix + '00'); // 00 is pushdata for no data
+    expect(resultPaymentId).toMatch(paymentIdRegex);
+  });
   it('Empty data no paymentId', () => {
-    const fullResult = parseOpReturnProps('', true)
-    expect(fullResult).toBe(allResultsPrefix + '00') // 00 is pushdata for no data
-  })
+    const fullResult = encodeOpReturnProps('', true);
+    expect(fullResult).toBe(allResultsPrefix + '00' + '00'); // the 00s are pushdata for the data and paymentId respectively
+  });
   it('Simple parse OpReturn 1', () => {
-    const fullResult = parseOpReturnProps('myCustomUserData') // 16 bytes
-    const resultPaymentId = fullResult?.slice(-paymentIdDigits)
-    const resultNoPaymentId = fullResult?.slice(0,-paymentIdDigits)
-    expect(resultNoPaymentId).toBe(allResultsPrefix +
-      '10' + // 16 in hex
-      '6d79437573746f6d5573657244617461')
-    expect(resultPaymentId?.length).toBe(paymentIdDigits)
-    expect(resultPaymentId).toMatch(paymentIdRegex)
-  })
+    const fullResult = encodeOpReturnProps('myCustomUserData'); // 16 bytes
+    const resultPaymentId = fullResult?.slice(-paymentIdDigits);
+    const resultNoPaymentId = fullResult?.slice(0, -paymentIdDigits);
+    expect(resultNoPaymentId).toBe(
+      allResultsPrefix +
+        '10' + // 16 in hex
+        '6d79437573746f6d5573657244617461',
+    );
+    expect(resultPaymentId?.length).toBe(paymentIdDigits);
+    expect(resultPaymentId).toMatch(paymentIdRegex);
+  });
   it('Simple parse OpReturn 1 no paymentId', () => {
-    const fullResult = parseOpReturnProps('myCustomUserData', true) // 16 bytes
-    expect(fullResult).toBe(allResultsPrefix +
-      '10' + // 16 in hex
-      '6d79437573746f6d5573657244617461')
-  })
+    const fullResult = encodeOpReturnProps('myCustomUserData', true); // 16 bytes
+    expect(fullResult).toBe(
+      allResultsPrefix +
+        '10' + // 16 in hex
+        '6d79437573746f6d5573657244617461' +
+        '00', // Empty paymentId pushdata
+    );
+  });
   it('Simple parse OpReturn 2', () => {
```

Comment:
`it('Encodes a v0 PayButton protocol tx with data of expected keyvalue shape and nonce')`